packages/venia-ui/lib/components/CartPage/PriceSummary/discountSummary.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import gql from 'fraql';
+import { Price } from '@magento/peregrine';
+
+const DEFAULT_AMOUNT = {
+    currency: 'USD',
+    value: 0
+};
+
+/**
+ * Reduces discounts array into a single amount.
+ *
+ * @param {Array} discounts
+ */
+const getDiscount = (discounts = []) => {
+    // discounts from data can be null
+    if (!discounts || !discounts.length) {
+        return DEFAULT_AMOUNT;
+    } else {
+        return {
+            currency: discounts[0].amount.currency,
+            value: discounts.reduce(
+                (acc, discount) => acc + discount.amount.value,
+                0
+            )
+        };
+    }
+};
+
+/**
+ * A component that renders the discount summary line item.
+ *
+ * @param {Object} props.classes
+ * @param {Object} props.data fragment response data
+ */
+const DiscountSummary = props => {
+    const { classes } = props;
+    const discount = getDiscount(props.data);
+
+    return discount.value ? (
+        <>
+            <span className={classes.lineItemLabel}>{'Discounts applied'}</span>
+            <span className={classes.price}>
+                {'(-'}
+                <Price
+                    value={discount.value}
+                    currencyCode={discount.currency}
+                />
+                {')'}
+            </span>
+        </>
+    ) : null;
+};
+
+DiscountSummary.fragments = {
+    discounts: gql`
+        fragment _ on CartPrices {
+            discounts {
```

Comment:
In 2.3.3 we have `discount`, which is a deprecated field in 2.3.4
In 2.3.4 we have `discounts`, which 2.3.3 does not have.

If we want to support both versions we will need to provide some way (build or runtime?) to determine which field is valid for the current backend.

Reply:
I am going to proceed with the assumption that we _don't_ need to support 2.3.3, so `discounts` is valid.